### PR TITLE
make SegmentMaxLifetime configurable

### PR DESCRIPTION
**Pull request by jiaoew1991**
issue: https://github.com/milvus-io/milvus/issues/16772
Signed-off-by: Enwei Jiao <jiaoew2011@gmail.com>

### CODE REVIEW

**Comment by xiaofan-luan** [comment]
what is SegmentMaxLifetime meaning for?

**Comment by xiaofan-luan** [comment]
hi @jiaoew1991, please sign off your DCO, https://github.com/milvus-io/milvus/blob/master/CONTRIBUTING.md

File: internal/util/paramtable/component_param.go [hunk 1]
```diff
@@ -915,6 +917,10 @@ func (p *dataCoordConfig) initSegAssignmentExpiration() {
 	p.SegAssignmentExpiration = p.Base.ParseInt64WithDefault("dataCoord.segment.assignmentExpiration", 2000)
 }
 
+func (p *dataCoordConfig) initSegmentMaxLifetime() {
+	p.SegmentMaxLifetime = time.Duration(p.Base.ParseInt64WithDefault("dataCoord.segment.maxLife", 24*60*60)) * time.Second
```
Comment (xiaofan-luan) [inline internal/util/paramtable/component_param.go:921]: put this dataCoord.segment.maxLife to milvus.yaml

Comment (jiaoew1991) [reply internal/util/paramtable/component_param.go:921]: done

**Comment by jiaoew1991** [comment]
rerun ut

**Comment by xiaofan-luan** [comment]
@jiaoew1991 Thanks for the contribution! Welcome to the community~

**Pull request merged**